explainer.md

```diff
@@ -215,6 +243,162 @@ third-party then signs either the exchanges in a package or the package as a
 whole using a certificate whose metadata reflects whichever property was
 reviewed for.
 
+## Loading sketch
+
+When an **embedder** prefetches or embeds an enveloped signed exchange, or a
+client navigates from the embedder to an enveloped signed exchange, the
+client goes through several steps to open the envelope and load the signed
+resource.
+
+### Fetch the physical URL
+
+The client won't know that a URL holds a signed exchange until it receives the
+`Content-Type` in the response, so the initial request is identical to any other
+request in the same context. It follows redirects, is constrained by the
+embedder's and any parent frame's Content Security Policy, and goes through the
+embedder's or the physical URL's Service Worker.
+
+Once the response comes back, its `Content-Type: application/signed-exchange`
+header tells the client that it represents a signed exchange, but it's initially
+treated like any other response: the Response object shown to the Service Worker
+is the bytes of the `application/signed-exchange` resource, not the response
+inside it, and the client follows the remaining steps only if the Service Worker
+responds with an encoded `application/signed-exchange` resource. The
+`application/signed-exchange` also participates in caches the same way as any
+other resource.
+
+### Fetch the certificate chain
+
+The client parses the beginning of the `application/signed-exchange` resource to
+extract the [`Signature`
+header](https://wicg.github.io/webpackage/draft-yasskin-http-origin-signed-responses.html#signature-header).
+Each signature in the `Signature` header has a `certUrl` field that identifies a
+certificate chain to use to validate the signature, and each certificate chain
+is fetched and validated.
+
+The request for this certificate chain is made without credentials and skips Service
+Workers, for compatibility with browser network stack architectures. (This may
+change if network stacks prove to be less of a problem than expected.) The
+certificate is cached as a normal HTTP request.
+
+The client checks the leaf certificate's hash against the `certSha256` field of
+the `Signature` header and then validates the certificate chain. Certificate
+validation is a call to an OS-controlled library and may make further URL
+requests that aren't under the browser's control and therefore use a separate,
+probably ephemeral, cookie jar, cache, and set of service workers.
+
+Each signature with a valid certificate chain is passed on to the next step.
+
+### Signature verification
+
+Once the certificates are validated and enough of the
+`application/signed-exchange` resource is received to parse the claimed
+signed-exchange headers, the client extracts the logical URL from those headers
+and then tries to find a valid signature over the headers that is trusted for
+the claimed origin. If it can't find any, it either
```

Comment:
Should it be clearer if we say 'If it fails to validate the signature' given that this talks about general validation failure cases (e.g. clock skew etc)?  (While I also found that we're using the text 'find a valid signature' throughout this change, so this is probably intentional?)

Reply:
My idea here is that signed exchanges can have several signatures, some of which are invalid for this client (but might be valid for some other client), some of which are valid but not origin-trusted, and some of which are both valid and origin-trusted. The basic verification process is a search through those signatures for one that's both valid and origin-trusted. Does that make sense? I definitely want to avoid implying that there's exactly one signature, but I'm happy to reword to make this clearer.